[serve] update get_application_url and tests

landscapepainter opened the pull request:
This PR replaces some of the manual string literals of urls within `test_api`, `test_deploy`, `test_deploy_2`, `test_deploy_app`, `test_failure` with `get_application_urls` and splits some of the tests into separate files.

edoakes committed:
[core] Unskip `test_placement_group_strict_spread` (#54381)

Reason it was skipped seems to no longer be relevant:
https://github.com/ray-project/ray/issues/38726

Let's see if it flakes...

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

alexeykudinkin committed:
[Data] Revisit async UDF handling in Ray Data (#54190)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

Currently async UDFs handling diverges substantially from sync UDF
handling:

1. Sync UDFs: process 1 batch at a time in a loop, yielding results
using generator (generator will block if executor is not keeping up to
taking up blocks from it)
2. Async UDFs: launches 1 async task per batch and runs all of them
*concurrently*, adding produced results into the queue, while elements
of that queue are yielded back as a result.

Currently, In async UDF scenario concurrency is not constrained at all
-- meaning that:

1. Async task for every batch will be launched ASAP
2. Output queue is not constrained and will be growing unbounded
(potentially resulting into OOMs)
3. No back-pressure is applied to throttle production irrespective of
the speed of the consuming Ray Task

Changes
---

This PR addresses all of the aforementioned problems by revisiting async
UDF handling protocol:

1. Limiting # of batches handled concurrently to no more than specified
`max_concurrency` (inside `ray_remote_args` or defaults of 2 for
map_batches and 16 for row-based operations)
2. Fixing handling of coroutines vs async-generators
3. Adding comprehensive tests

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: Alexey Kudinkin <ak@anyscale.com>
Signed-off-by: Linkun <github@lkchen.net>
Co-authored-by: Linkun <github@lkchen.net>
Signed-off-by: doyoung <doyoung@anyscale.com>

alexeykudinkin committed:
[Data] Revisit async UDF handling in Ray Data (#54190)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

Currently async UDFs handling diverges substantially from sync UDF
handling:

1. Sync UDFs: process 1 batch at a time in a loop, yielding results
using generator (generator will block if executor is not keeping up to
taking up blocks from it)
2. Async UDFs: launches 1 async task per batch and runs all of them
*concurrently*, adding produced results into the queue, while elements
of that queue are yielded back as a result.

Currently, In async UDF scenario concurrency is not constrained at all
-- meaning that:

1. Async task for every batch will be launched ASAP
2. Output queue is not constrained and will be growing unbounded
(potentially resulting into OOMs)
3. No back-pressure is applied to throttle production irrespective of
the speed of the consuming Ray Task

Changes
---

This PR addresses all of the aforementioned problems by revisiting async
UDF handling protocol:

1. Limiting # of batches handled concurrently to no more than specified
`max_concurrency` (inside `ray_remote_args` or defaults of 2 for
map_batches and 16 for row-based operations)
2. Fixing handling of coroutines vs async-generators
3. Adding comprehensive tests

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: Alexey Kudinkin <ak@anyscale.com>
Signed-off-by: Linkun <github@lkchen.net>
Co-authored-by: Linkun <github@lkchen.net>
Signed-off-by: doyoung <doyoung@anyscale.com>

kouroshHakha committed:
[data.llm] Return a batch of rows in the udf instead of row by row (#54329)

Signed-off-by: Kourosh Hakhamaneshi <kourosh@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

kouroshHakha committed:
[data.llm] Return a batch of rows in the udf instead of row by row (#54329)

Signed-off-by: Kourosh Hakhamaneshi <kourosh@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

Qiaolin-Yu committed:
[core][GPU Objects] Disable tensordict tests in macos ci (#54375)

tensordict requires macos Sequoia to run, which version is too high to run on ray's macos CI

Signed-off-by: doyoung <doyoung@anyscale.com>

Qiaolin-Yu committed:
[core][GPU Objects] Disable tensordict tests in macos ci (#54375)

tensordict requires macos Sequoia to run, which version is too high to run on ray's macos CI

Signed-off-by: doyoung <doyoung@anyscale.com>

khluu committed:
[release] Make KubeRay test run nightly (#54243)

Change KubeRay variant of this test to run nightly.
Please hold off on merging PR, @khluu needs to confirm if the job needs
to be deleted after each run (otherwise if the job stays on the cluster,
new jobs wouldn't run)

---------

Signed-off-by: Kevin H. Luu <kevin@anyscale.com>
Signed-off-by: kevin <kevin@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

khluu committed:
[release] Make KubeRay test run nightly (#54243)

Change KubeRay variant of this test to run nightly.
Please hold off on merging PR, @khluu needs to confirm if the job needs
to be deleted after each run (otherwise if the job stays on the cluster,
new jobs wouldn't run)

---------

Signed-off-by: Kevin H. Luu <kevin@anyscale.com>
Signed-off-by: kevin <kevin@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

elliot-barn committed:
updating raydepsets test job (#54387)

Updating test job for raydepsets 

- removing --only-tags and parallelism-per-worker flags

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>
Co-authored-by: Lonnie Liu <95255098+aslonnie@users.noreply.github.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

elliot-barn committed:
updating raydepsets test job (#54387)

Updating test job for raydepsets 

- removing --only-tags and parallelism-per-worker flags

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>
Co-authored-by: Lonnie Liu <95255098+aslonnie@users.noreply.github.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

aslonnie committed:
[ci] upgrade rayci version to 0.16.0 (#54392)

with some more features and goodies

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

aslonnie committed:
[ci] upgrade rayci version to 0.16.0 (#54392)

with some more features and goodies

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

alexeykudinkin committed:
[Data] Fixing `sort_benchmark` to avoid offsets overflows with Pyarrow (#54390)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

Addresses https://github.com/ray-project/ray/issues/53546

Currently, we're using unbounded size data type (binary) that internally
utilizes int32 offsets to index individual elements. Using int32 offsets
imposes a max limit on the cumulative size of the column being 2^31-1
bytes, which this tests consistently violates.

This change addresses that by replacing arbitrary length `binary` type
with fixed-size `binary` type that doesn't use int32 offsets (instead
just relying on int64 element counter)

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: Alexey Kudinkin <ak@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

alexeykudinkin committed:
[Data] Fixing `sort_benchmark` to avoid offsets overflows with Pyarrow (#54390)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

Addresses https://github.com/ray-project/ray/issues/53546

Currently, we're using unbounded size data type (binary) that internally
utilizes int32 offsets to index individual elements. Using int32 offsets
imposes a max limit on the cumulative size of the column being 2^31-1
bytes, which this tests consistently violates.

This change addresses that by replacing arbitrary length `binary` type
with fixed-size `binary` type that doesn't use int32 offsets (instead
just relying on int64 element counter)

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: Alexey Kudinkin <ak@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

aslonnie committed:
[data] dask: disable smoke test (#54411)

forgot to disable it last round

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

aslonnie committed:
[data] dask: disable smoke test (#54411)

forgot to disable it last round

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

aslonnie committed:
[ci] release tests: remove unused app configs (#54410)

many release configs are not used anywhere anymore. the tests are no
longer using the buildfarm to build the image

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

aslonnie committed:
[ci] release tests: remove unused app configs (#54410)

many release configs are not used anywhere anymore. the tests are no
longer using the buildfarm to build the image

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

harshit-anyscale committed:
migrate signature from _private to _common (#54357)

Fixes: https://github.com/ray-project/ray/issues/53478

- migrate ray._private.signature from `_private` to `_common`

Signed-off-by: doyoung <doyoung@anyscale.com>

harshit-anyscale committed:
migrate signature from _private to _common (#54357)

Fixes: https://github.com/ray-project/ray/issues/53478

- migrate ray._private.signature from `_private` to `_common`

Signed-off-by: doyoung <doyoung@anyscale.com>

dayshah committed:
[core] Delete skip_flaky_core_test_premerge (#54382)

`skip_flaky_core_test_premerge` was always skipping all the tests, not
just on premerge

Signed-off-by: dayshah <dhyey2019@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

dayshah committed:
[core] Delete skip_flaky_core_test_premerge (#54382)

`skip_flaky_core_test_premerge` was always skipping all the tests, not
just on premerge

Signed-off-by: dayshah <dhyey2019@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

owenowenisme committed:
[Core] Use smart pointer in logging.cc (#54351)

We shouldn't use raw pointer so I changed to smart pointer in loggin.cc.
Also use raw string literals to replace escape quotes in the same file
for better readability.


Signed-off-by: You-Cheng Lin (Owen) <mses010108@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

owenowenisme committed:
[Core] Use smart pointer in logging.cc (#54351)

We shouldn't use raw pointer so I changed to smart pointer in loggin.cc.
Also use raw string literals to replace escape quotes in the same file
for better readability.


Signed-off-by: You-Cheng Lin (Owen) <mses010108@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

dayshah committed:
[core] Inject reorder_wait_seconds for scheduling queue test (#54404)

This scheduling queue unit test would take >30 seconds before because
the default value of reorder_wait_seconds is 30. Now just injecting it
in one level higher so can inject it into the test as 1 second.

Signed-off-by: dayshah <dhyey2019@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

dayshah committed:
[core] Inject reorder_wait_seconds for scheduling queue test (#54404)

This scheduling queue unit test would take >30 seconds before because
the default value of reorder_wait_seconds is 30. Now just injecting it
in one level higher so can inject it into the test as 1 second.

Signed-off-by: dayshah <dhyey2019@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

edoakes committed:
[serve] Remove usage of `ray._private.state` (#54140)

Replace with `ray.util.state`.

There is one remaining:
```
(ray) eoakes@eoakes-CQTYX0Y0RV serve % grep -r "_private.state" *
_private/cluster_node_info_cache.py:            ray._private.state.available_resources_per_node()
```

---------

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

edoakes committed:
[serve] Remove usage of `ray._private.state` (#54140)

Replace with `ray.util.state`.

There is one remaining:
```
(ray) eoakes@eoakes-CQTYX0Y0RV serve % grep -r "_private.state" *
_private/cluster_node_info_cache.py:            ray._private.state.available_resources_per_node()
```

---------

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

edoakes committed:
[serve] Remove usage of `internal_api.memory_summary()` (#54417)

Replace with usage of the `list_objects` public state API.

---------

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

edoakes committed:
[serve] Remove usage of `internal_api.memory_summary()` (#54417)

Replace with usage of the `list_objects` public state API.

---------

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

srinathk10 committed:
[Data] Add `pin_memory` to `iter_torch_batches` (#53792)

Add `pin_memory` to `iter_torch_batches`
- Whether to pin the memory of the tensors. Defaults to False.

---------

Signed-off-by: Srinath Krishnamachari <srinath.krishnamachari@anyscale.com>
Signed-off-by: srinathk10 <68668616+srinathk10@users.noreply.github.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

srinathk10 committed:
[Data] Add `pin_memory` to `iter_torch_batches` (#53792)

Add `pin_memory` to `iter_torch_batches`
- Whether to pin the memory of the tensors. Defaults to False.

---------

Signed-off-by: Srinath Krishnamachari <srinath.krishnamachari@anyscale.com>
Signed-off-by: srinathk10 <68668616+srinathk10@users.noreply.github.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

edoakes committed:
[core] Skip `test_owner_assign_inner_object` on Windows (#54383)

Test is flaky on Windows. `_owner=` arg is experimental and will be
deprecated/removed soon, no need to spend time debugging this on
Windows.

Closes https://github.com/ray-project/ray/issues/54333

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

edoakes committed:
[core] Skip `test_owner_assign_inner_object` on Windows (#54383)

Test is flaky on Windows. `_owner=` arg is experimental and will be
deprecated/removed soon, no need to spend time debugging this on
Windows.

Closes https://github.com/ray-project/ray/issues/54333

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

noemotiovon committed:
[Refactor]Rename NCCL-related items to comm_backend (#51061)

## Why are these changes needed?

### Background
This PR is a follow-up to
[#51032](https://github.com/ray-project/ray/pull/51032), which
introduced multi-device support in the Compiled Graph by leveraging
CUDA's NCCL backend for efficient out-of-band tensor communication.

While the current implementation is tightly coupled with NCCL and CUDA,
the Compiled Graph runtime is now ready to support a broader spectrum of
device types and collective communication backends (e.g., HCCL, RCCL).

### What This PR Does?
To enable extensibility and backend-agnostic design, this PR introduces
the following core changes:

Refactored NCCL-specific naming and APIs
NCCL-related modules, classes, and function names have been generalized
to eliminate hardcoded CUDA/NCCL assumptions.

Introduced a pluggable communication backend interface
A unified abstraction layer is added to decouple collective
communication logic from any specific implementation. This makes it
easier to support alternative collective libraries and device types in
the future.

This refactor does not alter the existing behavior of NCCL-based
Compiled Graph execution. All current workflows using CUDA+NCCL continue
to function as before.


## Related issue number
https://github.com/ray-project/ray/issues/51574

<!-- For example: "Closes #1234" -->

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: noemotiovon <757486878@qq.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

noemotiovon committed:
[Refactor]Rename NCCL-related items to comm_backend (#51061)

## Why are these changes needed?

### Background
This PR is a follow-up to
[#51032](https://github.com/ray-project/ray/pull/51032), which
introduced multi-device support in the Compiled Graph by leveraging
CUDA's NCCL backend for efficient out-of-band tensor communication.

While the current implementation is tightly coupled with NCCL and CUDA,
the Compiled Graph runtime is now ready to support a broader spectrum of
device types and collective communication backends (e.g., HCCL, RCCL).

### What This PR Does?
To enable extensibility and backend-agnostic design, this PR introduces
the following core changes:

Refactored NCCL-specific naming and APIs
NCCL-related modules, classes, and function names have been generalized
to eliminate hardcoded CUDA/NCCL assumptions.

Introduced a pluggable communication backend interface
A unified abstraction layer is added to decouple collective
communication logic from any specific implementation. This makes it
easier to support alternative collective libraries and device types in
the future.

This refactor does not alter the existing behavior of NCCL-based
Compiled Graph execution. All current workflows using CUDA+NCCL continue
to function as before.


## Related issue number
https://github.com/ray-project/ray/issues/51574

<!-- For example: "Closes #1234" -->

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: noemotiovon <757486878@qq.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

ryanaoleary committed:
[Autoscaler][V2] Use running node instances to rate-limit upscaling (#50414)

This PR changes the v2 autoscaler scale up logic to use
`IMInstance.RAY_RUNNING` (rather than `IMInstance.REQUESTED` + nodes
with an `IMInstance.cloud_instance_id`) when computing the number of
nodes to launch per node type. This is to maintain consistency with how
`upscaling_speed` is handled by the V1 autoscaler and described in the
[Ray
docs](https://docs.ray.io/en/latest/cluster/vms/user-guides/configuring-autoscaling.html#upscaling-and-downscaling-speed),
i.e. the factor by which to scale the number of nodes in the cluster,
with a minimum of 5.

---------

Signed-off-by: Ryan O'Leary <ryanaoleary@google.com>
Co-authored-by: Kai-Hsun Chen <kaihsun@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

ryanaoleary committed:
[Autoscaler][V2] Use running node instances to rate-limit upscaling (#50414)

This PR changes the v2 autoscaler scale up logic to use
`IMInstance.RAY_RUNNING` (rather than `IMInstance.REQUESTED` + nodes
with an `IMInstance.cloud_instance_id`) when computing the number of
nodes to launch per node type. This is to maintain consistency with how
`upscaling_speed` is handled by the V1 autoscaler and described in the
[Ray
docs](https://docs.ray.io/en/latest/cluster/vms/user-guides/configuring-autoscaling.html#upscaling-and-downscaling-speed),
i.e. the factor by which to scale the number of nodes in the cluster,
with a minimum of 5.

---------

Signed-off-by: Ryan O'Leary <ryanaoleary@google.com>
Co-authored-by: Kai-Hsun Chen <kaihsun@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

aslonnie committed:
[wheel] remove pyarrow <18 restriction (#54405)

the restriction should be on installation time rather than inside the
wheel. this extra constraint sometimes can unnecessarily mislead
depedency manager's resolution algorithms

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

aslonnie committed:
[wheel] remove pyarrow <18 restriction (#54405)

the restriction should be on installation time rather than inside the
wheel. this extra constraint sometimes can unnecessarily mislead
depedency manager's resolution algorithms

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

zcin committed:
[serve] bump timeout for test_proxy (#54426)

Move `test_proxy` to `medium` size test.

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

zcin committed:
[serve] bump timeout for test_proxy (#54426)

Move `test_proxy` to `medium` size test.

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

aslonnie committed:
[deps] sync ml byod deps with global lock file (#54424)

update the versions of more-itertools and stuff

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

aslonnie committed:
[deps] sync ml byod deps with global lock file (#54424)

update the versions of more-itertools and stuff

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

aslonnie committed:
[ci] add workspace status script (#54398)

and use it to reject '@' character in `HOME` and `USER`

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

aslonnie committed:
[ci] add workspace status script (#54398)

and use it to reject '@' character in `HOME` and `USER`

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

zcin committed:
[serve] add request metadata to can_accept_request (#54429)

Add `request_metadata` as a parameter for future use.

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

zcin committed:
[serve] add request metadata to can_accept_request (#54429)

Add `request_metadata` as a parameter for future use.

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

raulchen committed:
[data] Extract backpressure-related code from ResourceManager as a policy (#54376)

* Extract backpressure related methods (`can_submit_new_tasks` and
`max_task_output_bytes_to_read`) from ResourceManager, as a standalone
policy `ResourceBudgetBackpressurePolicy`)
* Report task_output_backpressure_time metric.

---------

Signed-off-by: Hao Chen <chenh1024@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

raulchen committed:
[data] Extract backpressure-related code from ResourceManager as a policy (#54376)

* Extract backpressure related methods (`can_submit_new_tasks` and
`max_task_output_bytes_to_read`) from ResourceManager, as a standalone
policy `ResourceBudgetBackpressurePolicy`)
* Report task_output_backpressure_time metric.

---------

Signed-off-by: Hao Chen <chenh1024@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

TimothySeah committed:
[train][doc] ray.train.report api docs should mention optional checkpoint_dir_name (#54391)


---------

Signed-off-by: Timothy Seah <tseah@anyscale.com>
Signed-off-by: Timothy Seah <timothy.seah777@yahoo.com>
Co-authored-by: Timothy Seah <tseah@anyscale.com>
Co-authored-by: Justin Yu <justinvyu@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

TimothySeah committed:
[train][doc] ray.train.report api docs should mention optional checkpoint_dir_name (#54391)


---------

Signed-off-by: Timothy Seah <tseah@anyscale.com>
Signed-off-by: Timothy Seah <timothy.seah777@yahoo.com>
Co-authored-by: Timothy Seah <tseah@anyscale.com>
Co-authored-by: Justin Yu <justinvyu@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

landscapepainter committed:
fix

Signed-off-by: doyoung <doyoung@anyscale.com>

landscapepainter committed:
fix

Signed-off-by: doyoung <doyoung@anyscale.com>

vigneshka committed:
Update google tag container id (#54444)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

We don't have access to the previous tag manager and can't edit event
collection. This new one works



<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

<!-- Please give a short summary of the change and the problem this
solves. -->

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

Signed-off-by: Vignesh Hirudayakanth <vigneshkanth@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

vigneshka committed:
Update google tag container id (#54444)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

We don't have access to the previous tag manager and can't edit event
collection. This new one works



<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

<!-- Please give a short summary of the change and the problem this
solves. -->

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

Signed-off-by: Vignesh Hirudayakanth <vigneshkanth@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

bveeramani committed:
[Data] Avoid OOMs with `read_json(..., lines=True)` (#54436)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

<!-- Please give a short summary of the change and the problem this
solves. -->

`read_json(..., lines=True)` uses pandas to read batches of data. The
issue is that the batch size is always set to 10000 rows per batch, and
this can cause OOMs if each line is large (e.g., a line might contain
bytes for a 8 MiB image).

To avoid OOMs, this PR updates the implementation to sample a row and
infer the appropriate batch size.

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: Balaji Veeramani <bveeramani@berkeley.edu>
Signed-off-by: doyoung <doyoung@anyscale.com>

bveeramani committed:
[Data] Avoid OOMs with `read_json(..., lines=True)` (#54436)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

<!-- Please give a short summary of the change and the problem this
solves. -->

`read_json(..., lines=True)` uses pandas to read batches of data. The
issue is that the batch size is always set to 10000 rows per batch, and
this can cause OOMs if each line is large (e.g., a line might contain
bytes for a 8 MiB image).

To avoid OOMs, this PR updates the implementation to sample a row and
infer the appropriate batch size.

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: Balaji Veeramani <bveeramani@berkeley.edu>
Signed-off-by: doyoung <doyoung@anyscale.com>

zcin commented on python/ray/serve/tests/test_api.py:
```diff
@@ -433,14 +433,15 @@ def g():
     # make sure no affect to app_g
     assert g_handle.remote().result() == "got g"
-    url = f"{get_application_url(app_name='app_g')}"
+    url = "http://localhost:8000/app_g"
```
why does this need to go through 8000?

landscapepainter replied:
@zcin Thanks for the catch. It should grab the url through get_application_url, not the fixed proxy url. Fixed at [cdf5d03](https://github.com/ray-project/ray/pull/54449/commits/cdf5d037520089e2b249c8adc0ae355a75a4e8d5)

zcin commented on python/ray/serve/_private/test_utils.py:
```diff
@@ -708,23 +708,30 @@ def tlog(s: str, level: str = "INFO"):
 def get_application_urls(
     protocol: Union[str, RequestProtocol] = RequestProtocol.HTTP,
     app_name: str = SERVE_DEFAULT_APP_NAME,
+    route_prefix: str = None,
```
is this used?

landscapepainter replied:
@zcin It's not used. Removed.

zcin commented on python/ray/serve/tests/test_deploy_2.py:
```diff
@@ -239,7 +243,7 @@ def check_fail():
         return True
 
-    wait_for_condition(check_fail, timeout=15)
+    wait_for_condition(check_fail, timeout=30)
```
why does this need to be bumped? was it flaky before?

landscapepainter replied:
@zcin It's not used. Removed.

landscapepainter replied:
Correct.

srinathk10 committed:
Fix test_json CI failures (#54352)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

<!-- Please give a short summary of the change and the problem this
solves. -->

Fix test_json CI failures 

- Reduce the file size to KBs (instead of MBs) for testing file size
larger than block size in `test_json_read_file_larger_than_block_size`.

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: Srinath Krishnamachari <srinath.krishnamachari@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

srinathk10 committed:
Fix test_json CI failures (#54352)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

<!-- Please give a short summary of the change and the problem this
solves. -->

Fix test_json CI failures 

- Reduce the file size to KBs (instead of MBs) for testing file size
larger than block size in `test_json_read_file_larger_than_block_size`.

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: Srinath Krishnamachari <srinath.krishnamachari@anyscale.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

zcin committed:
[serve] skip test_proxy_disconnect_metrics on windows (#54441)

Skip `test_proxy_disconnect_metrics` on windows because it is flaky.

---------

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

zcin committed:
[serve] skip test_proxy_disconnect_metrics on windows (#54441)

Skip `test_proxy_disconnect_metrics` on windows because it is flaky.

---------

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>
Signed-off-by: doyoung <doyoung@anyscale.com>

landscapepainter committed:
remote route_prefix from get_application_urls

Signed-off-by: doyoung <doyoung@anyscale.com>

landscapepainter committed:
remote route_prefix from get_application_urls

Signed-off-by: doyoung <doyoung@anyscale.com>

landscapepainter committed:
update test_delete_application

Signed-off-by: doyoung <doyoung@anyscale.com>

landscapepainter committed:
update test_delete_application

Signed-off-by: doyoung <doyoung@anyscale.com>

landscapepainter committed:
format

Signed-off-by: doyoung <doyoung@anyscale.com>

landscapepainter committed:
format

Signed-off-by: doyoung <doyoung@anyscale.com>

landscapepainter committed:
Merge branch 'master' into update-get-application-url-and-tests

landscapepainter committed:
format

landscapepainter committed:
format

Signed-off-by: doyoung <doyoung@anyscale.com>

landscapepainter committed:
Merge branch 'update-get-application-url-and-tests' of https://github.com/landscapepainter/ray into update-get-application-url-and-tests

landscapepainter committed:
Merge branch 'update-get-application-url-and-tests' of https://github.com/landscapepainter/ray into update-get-application-url-and-tests

abrarsheikh reviewed (approved):
looks good

landscapepainter committed:
Merge branch 'master' into update-get-application-url-and-tests

landscapepainter committed:
add comment for test

Signed-off-by: doyoung <doyoung@anyscale.com>

zcin commented:
@landscapepainter please add a pr description
